docs/BestPractices.md

```diff
@@ -0,0 +1,44 @@
+#  AWS Java SDK 2.x Best Practices
+
+Here are the best practices of using AWS Java SDK 2.x.
+
+### Reuse SDK client if possible
+
+Each SDK client maintains many resources such as connections and threads, so it is recommended to share a single instance of the client to avoid the overhead of resources management. All SDK clients are thread safe. 
+
+If it is not desirable to share a client instance, call `client.close()` to release the resources once the client is not needed.
+
+### Close input stream after read
```

Comment:
This sounds a little ambiguous. How about "Close InputStreams from client operations".

We might also what to add a comment about reading the stream completely, otherwise the connection can't be reused